pkg/volume/flocker/plugin_test.go

```diff
@@ -28,15 +30,17 @@ import (
 
 const pluginName = "kubernetes.io/flocker"
 
-func newInitializedVolumePlugMgr() volume.VolumePluginMgr {
+func newInitializedVolumePlugMgr(t *testing.T) (volume.VolumePluginMgr, string) {
 	plugMgr := volume.VolumePluginMgr{}
-	plugMgr.InitPlugins(ProbeVolumePlugins(), volume.NewFakeVolumeHost("/foo/bar", nil, nil))
-	return plugMgr
+	dir, err := ioutil.TempDir("", "flocker")
+	assert.NoError(t, err)
+	plugMgr.InitPlugins(ProbeVolumePlugins(), volume.NewFakeVolumeHost(dir, nil, nil))
+	return plugMgr, dir
```

Comment:
Honestly I'd rather return the dir here and just ignore it when not needed
vs. incorporating logic into the unit test that reaches into the bowels of
a structure to extract the path some other way. This seems much more
straightforward.

On Wed, Oct 7, 2015 at 10:44 AM, Alexandre González <
notifications@github.com> wrote:

> In pkg/volume/flocker/plugin_test.go
> <https://github.com/kubernetes/kubernetes/pull/15158#discussion_r41398502>
> :
>
> >  	plugMgr := volume.VolumePluginMgr{}
> > -	plugMgr.InitPlugins(ProbeVolumePlugins(), volume.NewFakeVolumeHost("/foo/bar", nil, nil))
> > -	return plugMgr
> > +	dir, err := ioutil.TempDir("", "flocker")
> > +	assert.NoError(t, err)
> > +	plugMgr.InitPlugins(ProbeVolumePlugins(), volume.NewFakeVolumeHost(dir, nil, nil))
> > +	return plugMgr, dir
>
> I wonder if there is a better way of getting this dir than returning it
> here (it's not being used in 3 out of 4 functions). Perhaps with the
> builder created in line 212? There is no problem doing the cleanup there
> because the function SetupAt was not called yet.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/15158/files#r41398502>.
>